diff -rNU3 -xCVS dist.orig/bfd/config.bfd dist/bfd/config.bfd
--- dist.orig/bfd/config.bfd	Fri Feb  8 14:19:35 2013
+++ dist/bfd/config.bfd	Fri Feb  8 14:52:30 2013
@@ -281,7 +281,7 @@
     ;;
   arm-*-elf | arm-*-freebsd* | arm*-*-linux-* | arm*-*-conix* | \
   arm*-*-uclinux* | arm-*-kfreebsd*-gnu | \
-  arm*-*-eabi* )
+  arm*-*-eabi* | arm*-*-minix*)
     targ_defvec=bfd_elf32_littlearm_vec
     targ_selvecs=bfd_elf32_bigarm_vec
     ;;
@@ -631,7 +631,7 @@
     targ_selvecs="bfd_elf32_i386_freebsd_vec i386coff_vec i386pei_vec x86_64pei_vec bfd_elf32_i386_vec bfd_elf64_x86_64_vec bfd_elf64_l1om_vec bfd_elf64_l1om_freebsd_vec"
     want64=true
     ;;
-  x86_64-*-netbsd* | x86_64-*-openbsd*)
+  x86_64-*-netbsd* | x86_64-*-openbsd* | x86_64-*-minix*)
     targ_defvec=bfd_elf64_x86_64_vec
     targ_selvecs="bfd_elf32_i386_vec i386netbsd_vec i386coff_vec i386pei_vec x86_64pei_vec bfd_elf64_l1om_vec"
     want64=true
@@ -696,6 +696,9 @@
     ;;
   i[3-7]86-none-*)
     targ_defvec=i386coff_vec
+    ;;
+  i[3-7]86-*-minix*)
+    targ_defvec=bfd_elf32_i386_vec
     ;;
   i[3-7]86-*-aout* | i[3-7]86*-*-vsta*)
     targ_defvec=i386aout_vec
diff -rNU3 -xCVS dist.orig/config.guess dist/config.guess
--- dist.orig/config.guess	Fri Feb  8 14:19:36 2013
+++ dist/config.guess	Fri Feb  8 14:54:57 2013
@@ -855,8 +855,11 @@
 	# other systems with GNU libc and userland
 	echo ${UNAME_MACHINE}-unknown-`echo ${UNAME_SYSTEM} | sed 's,^[^/]*/,,' | tr '[A-Z]' '[a-z]'``echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'`-gnu
 	exit ;;
-    i*86:Minix:*:*)
+    i*86:Minix:*:* | x86_64:Minix:*:*)
 	echo ${UNAME_MACHINE}-pc-minix
+	exit ;;
+    *arm*:Minix:*:*)
+	echo ${UNAME_MACHINE}-elf32-minix
 	exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
diff -rNU3 -xCVS dist.orig/gas/configure.tgt dist/gas/configure.tgt
--- dist.orig/gas/configure.tgt	Fri Apr  1 14:02:42 2011
+++ dist/gas/configure.tgt	Fri Feb  8 14:59:02 2013
@@ -425,7 +425,7 @@
   *-*-generic)				fmt=generic ;;
   *-*-xray | *-*-hms)			fmt=coff ;;
   *-*-sim)				fmt=coff ;;
-  *-*-elf | *-*-sysv4*)			fmt=elf ;;
+  *-*-elf | *-*-sysv4* | *-*-minix*)	fmt=elf ;;
   *-*-solaris*)				fmt=elf em=solaris ;;
   *-*-aros*)				fmt=elf em=linux ;;
   *-*-vxworks* | *-*-windiss)		fmt=elf em=vxworks ;;
diff -rNU3 -xCVS dist.orig/ld/configure.tgt dist/ld/configure.tgt
--- dist.orig/ld/configure.tgt	Fri Feb  8 14:19:37 2013
+++ dist/ld/configure.tgt	Fri Feb  8 19:03:56 2013
@@ -71,7 +71,7 @@
 arm-*-openbsd*)		targ_emul=armnbsd ;;
 arm-*-rtems*)		targ_emul=armelf ;;
 armeb-*-elf)		targ_emul=armelfb ;;
-arm-*-elf | arm*-*-eabi*)
+arm-*-elf | arm*-*-eabi* | arm*-*-minix*)
 	  		targ_emul=armelf ;;
 arm*-*-symbianelf*)     targ_emul=armsymbian;;
 arm-*-kaos*)		targ_emul=armelf ;;
@@ -272,6 +272,9 @@
 i[3-7]86-*-ptx*)	targ_emul=i386coff ;;
 i[3-7]86-*-mach*)	targ_emul=i386mach ;;
 i[3-7]86-*-gnu*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-minix*)	targ_emul=elf_i386 ;;
+x86_64-*-minix*)	targ_emul=elf_x86_64
+			targ_extra_emuls="elf_i386" ;;
 i[3-7]86-*-msdos*)	targ_emul=i386msdos; targ_extra_emuls=i386aout ;;
 i[3-7]86-*-moss*)	targ_emul=i386moss; targ_extra_emuls=i386msdos ;;
 i[3-7]86-*-winnt*)	targ_emul=i386pe ;
